README.md

```diff
@@ -27,83 +38,56 @@ const app = new App({
-   inside the event envelope). For a block action or legacy action, it will be the action inside the `actions` array.
-   The same object will also be available with the specific name for the kind of payload it is. For example, for an
-   event from a block action, you can use the `payload` and `action` arguments interchangeably. **The easiest way to
-   understand what's in a payload is to simply log it**, or otherwise [use TypeScript](#using-typescript).
+Most of the app's functionality will be inside listener functions (the `fn` parameters above). These functions are called with a set of arguments.
 
-*  `say` - A function to respond to an incoming event. This argument is only available when the listener is triggered
-   for event that contains a `channel_id` (including `message` events). Call this function to send a message back to the
-   same channel as the incoming event. It accepts both simple strings (for plain messages) and objects (for complex
-   messages, including blocks or attachments). `say` returns a promise that will resolve with a
-   [response](https://api.slack.com/methods/chat.postMessage) from `chat.postMessage`.
+| Argument  | Description  |
+| :---: | :--- |
+| `payload` | Contents of the incoming event. The payload structure depends on the listener. For example, for an Events API event, `payload` will be the [event type structure](https://api.slack.com/events-api#event_type_structure). For a block action, it will be the action from within the `actions` array. The `payload` object is also accessible via the alias corresponding to the listener (`message`, `event`, `action`, `shortcut`, `command`, or `options`). For example, if you were building a `message()` listener, you could use the `payload` and `message` arguments interchangably. **An easy way to understand what's in a payload is to log it**, or [use TypeScript](https://slack.dev/bolt-js/tutorial/using-typescript). |
+| `say` | Function to send a message to the channel associated with the incoming event. This argument is only available when the listener is triggered for events that contain a `channel_id` (the most common being `message` events). `say` accepts simple strings (for plain-text messages) and objects (for messages containing blocks). `say` returns a promise that will resolve with a [`chat.postMessage` response](https://api.slack.com/methods/chat.postMessage).
+| `ack` | Function that **must** be called to acknowledge that an incoming event was received by your app. `ack` exists for all actions, shortcuts, view, slash command and options requests. `ack` returns a promise that resolves when complete. Read more in [Acknowledging events](#acknowledging-events)
+| `client` | Web API client that uses the token associated with that event. For single-workspace installations, the token is provided to the constructor. For multi-workspace installations, the token is returned by the authorization function.
```

Comment:
This is not a strong suggestion at all but changing this way may make it more specific.

```suggestion
| `client` | Web API client that uses the token associated with that event. For single-workspace installations, the token is provided to the constructor. For multi-workspace installations, the token is returned by the `authorize` function.
```